ha/core/controllers/pcs/pcs_controller.py

```diff
@@ -181,4 +185,47 @@ def _get_filtered_nodes(self, status: list) -> list:
         for node in node_status.keys():
             if node_status[node] in status:
                 nodelist.append(node)
-        return nodelist
\ No newline at end of file
+        return nodelist
+
+    def is_valid_node_id(self, node_id) -> bool:
+        '''
+           Checks if node id gets resolved to some IP address or not
+           Returns: bool
+           Exception: socket.gaierror, socket.herror
+        '''
+        # TODO: change this logic and validate the node_id from the
+        # list coming from system health
+        ip_validator_regex = "^((25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])$"
+
+        # node_id can be passed as IP address or FQDN or
+        # some random number or just sequence of chars
+
+        # first seperate the string from dots.
+        splitted_node_id = node_id.replace('.', '')
+
+        # If string only contains numbers, means it can be just
+        # random number or can be an IP address. So, IP address
+        # validation can be done. else for random number, exception will be
+        # raised
+        if re.search('^[0-9]*$', splitted_node_id):
+            if re.search(ip_validator_regex, node_id):
+                return True
+            raise HAClusterCLIError(f'{node_id} is not a valid node_id')
+        # else it can be combination of chars and numbers means hostname or just a
+        # random meaningless string
+        else:
+            try:
+                socket.gethostbyname(node_id)
+            except Exception as err:
+                raise HAClusterCLIError(f'{node_id} not a valid node_id: {err}')
+        return True
+
+    def is_node_exists(self, node_id):
```

Comment:
Think of better name for is_node_exists - is_node_in_cluster or does_node_exist would be better.
Also, this should be a private function.
Another thing, there is already a function with name _get_node_list. Get the node list and see if it's there in pcs cluster or not. In your implementation, you are going to _confstore which is also fine but _get_node_list gets the list from pacemaker itself and that would be better.

Reply:
Function "is_node_exists" changed as per comment.